pkg/controller/admissionchecks/provisioning/cache.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"sync"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+type cache struct {
+	lock    sync.RWMutex
+	checks  map[string]*kueue.AdmissionCheck
+	configs map[string]*kueue.ProvisioningRequestConfig
+}
+
+func parametersRefValid(params *kueue.AdmissionCheckParametersReference) bool {
+	if params == nil {
+		return false
+	}
+	return params.Kind == ConfigKind && params.APIGroup == kueue.GroupVersion.Group && params.Name != ""
+}
+
+func (c *cache) AddOrUpdateCheck(check *kueue.AdmissionCheck) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.checks[check.Name] = check
+}
+
+func (c *cache) RemoveCheck(check *kueue.AdmissionCheck) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	delete(c.checks, check.Name)
+}
+
+func (c *cache) AddOrUpdateConfig(config *kueue.ProvisioningRequestConfig) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.configs[config.Name] = config
+}
+
+func (c *cache) RemoveConfig(config *kueue.ProvisioningRequestConfig) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	delete(c.configs, config.Name)
+}
+
+// FilterChecks - returns a list of check names controlling one of the provided states
+// and present in the cache.
+func (c *cache) FilterChecks(states []kueue.AdmissionCheckState) []string {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	var retActive []string
+	for _, state := range states {
+		if ac, found := c.checks[state.Name]; found {
+			retActive = append(retActive, ac.Name)
+		}
+	}
+	return retActive
+}
+
+// GetConfigForCheck - get the config used by the check identified by its name,
+// nil is return if the config is missing or improperly configured (the check is not active).
+func (c *cache) GetConfigForCheck(name string) *kueue.ProvisioningRequestConfig {
```

Comment:
Please clarify in the comment whose name is used in params. AdmissionCheck or ProvisioningRequestConfig.